vncdotool/command.py

```diff
@@ -277,22 +281,27 @@ def setup_logging(options):
     PythonLoggingObserver().start()
 
 
-def parse_host(server):
+def parse_server(server):
     split = server.split(':')
 
     if not split[0]:
         host = '127.0.0.1'
     else:
         host = split[0]
 
+    if os.path.exists(host):
```

Comment:
What happens if its a unix socket style path but the socket doesn't exist?

Reply:
You get the following error which is the same error you would get with the current logic if instead of passing a valid hostname you pass a path.

```
VNCDoException: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.DNSLookupError'>: DNS lookup failed: address '/foo/bar/baz' not found: [Errno -2] Name or service not known.
]
```

There is not a proper way to distinguish between a hostname and a relative path at a string level. Someone could for example input the following relative path: `example.com/unix.skt`.

Only working approach would be calling [getaddrinfo](https://docs.python.org/3/library/socket.html#socket.getaddrinfo) if `os.path.exists` returns `False` to verify it's a legitimate hostname. Unfortunately, `getaddrinfo` will trigger a DNS resolution which might lead to few unexpected behaviours.

Reply:
good point, yuck. We could bake in a bunch of logic to guess the type but as you say ultimately it will fail.  The best we could do it provide a nicer error message or document the behaviour but that shouldn't hold up this PR.